### DISCUSSION TITLE
Statistics explanations

### DISCUSSION THREAD

**Discussion by nomantis** [ccache/ccache#1271] [Questions & answers] [answered]
Hi all!

I've been trying to find what the statistics mean that are shown in `--print-stats`, but they don't seem to be documented. Could somebody explain them to me? Maybe this could be documented, so if these are collected for analysis, we have a common understanding of what they mean.

```
autoconf_test
bad_compiler_arguments
bad_output_file
cache_miss
cache_size_kibibyte
called_for_link
called_for_preprocessing
cleanups_performed
compile_failed
compiler_check_failed
compiler_produced_empty_output
compiler_produced_no_output
compiler_produced_stdout
could_not_find_compiler
could_not_use_modules
could_not_use_precompiled_header
direct_cache_hit
direct_cache_miss
error_hashing_extra_file
files_in_cache
internal_error
local_storage_hit
local_storage_miss
local_storage_read_hit
local_storage_read_miss
local_storage_write
missing_cache_file
multiple_source_files
no_input_file
output_to_stdout
preprocessed_cache_hit
preprocessed_cache_miss
preprocessor_error
recache
remote_storage_error
remote_storage_hit
remote_storage_miss
remote_storage_read_hit
remote_storage_read_miss
remote_storage_timeout
remote_storage_write
stats_updated_timestamp
stats_zeroed_timestamp
unsupported_code_directive
unsupported_compiler_option
unsupported_environment_variable
unsupported_source_language
```

**Comment by jrosdahl** [comment]
You are right, the identifiers you mentioned are not explicitly referenced in the documentation. However, their corresponding human-readable descriptions can be found in the  _[Cache statistics](https://ccache.dev/manual/4.8.html#_cache_statistics)_ section of the manual. To map these descriptions to their respective identifiers, refer to the [`k_statistics_field` array](https://github.com/ccache/ccache/blob/fe03859ebe1e5fdf87b02fba3d70f5854124d453/src/core/Statistics.cpp#L69-L130) located in `src/core/Statistics.cpp`. For the remaining statistics fields without descriptions, they generally represent self-explanatory counters, such as different hit/miss counters.

**Reply by nomantis** [reply]
Thank you for the links, I have already pieced these together before asking the question. The reason I asked is that the stats are not self-explanatory in my eyes. I could see multiple interpretations of them just by searching around the web. This is why I would suggest making them concrete, either by comments in the code next to the statistics or by adding them to the documentation you've linked. I'm leaning towards comments in the code, since then there isn't a need to update a separate documentation if there are any changes in the stats.

**Reply by jrosdahl** [reply, marked as answer]
I made an attempt in e08efb42c194.

**Reply by nomantis** [reply]
Thank you! This goes a long way in making conversations and dashboards about these stats productive! 🙇🏻
